onnx/defs/schema.cc

```diff
@@ -381,6 +331,27 @@ void OpSchema::Verify(const NodeProto& node) const {
   // Phew. All verifications passed.
 }
 
+void OpSchema::VerifyInputNum(int input_num, const std::string& node_info) const {
+  if (input_num < min_input_ || input_num > max_input_) {
+    fail_check(node_info, " has input size ", input_num, " not in range [min=", min_input_, ", max=", max_input_, "].");
+  }
+
+  if (!num_inputs_allowed_(input_num)) {
+    fail_check(node_info, " has input size ", input_num, " not in allowed input sizes.");
+  }
+}
+
+void OpSchema::VerifyOutputNum(int output_num, const std::string& node_info) const {
+  if (output_num < min_output_ || output_num > max_output_) {
+    fail_check(
+        node_info, " has output size ", output_num, " not in range [min=", min_output_, ", max=", max_output_, "].");
+  }
+
+  if (!num_outputs_allowed_(output_num)) {
+    fail_check(node_info, " has output size ", output_num, " not in allowed output sizes.");
+  }
```

Comment:
Should we handle when an output is variadic as well? For example, https://onnx.ai/onnx/operators/onnx__Split.html

Or is it handled by max_output_ already?

Reply:
The `output_num` will be set in [`OpSchema::Finalize`](https://github.com/onnx/onnx/blob/main/onnx/defs/schema.cc#L956), and the function is typically invoked before registration.